astropy/io/fits/hdu/compressed/_tiled_compression.py

```diff
@@ -465,7 +467,11 @@ def decompress_image_data_section(
             if zblank is not None:
                 if not tile_data.flags.writeable:
                     tile_data = tile_data.copy()
-                tile_data[blank_mask] = np.nan
+                if zbitpix > 0:
+                    blank = zblank_header
```

Comment:
should probably be `= zblank`, which can be `= zblank_header` or `= zblank_column[row_index]` if ZBLANK is specified per tile (see above).

Reply:
No, it cannot be `zblank_header` because that can be different for every tile.  Once the image has been uncompressed, there must be a single `BLANK` value that is used for the entire image.  There is no longer any memory of the tiles, so using different blank values for different tiles is impossible.

Reply:
Right :+1: 
But then we should make sure that the `blank` value here (`zblank_header` for non floating data) is defined. If `ZBLANK`is specified as a column and not in the header, currently you will get None...

Reply:
There is code in `compressed.py` that adds a `BLANK` keyword to the header (with a warning) for the only case where `BLANK` and `ZBLANK` are missing but `BLANK` is still needed.  (That can occur if there is a `ZBLANK` column in the compressed data table.)  So it is not necessary to test further for it here. If we added redundant code here, I think the test coverage would fail since there would be some code lines that cannot be reached by any tests.

Reply:
Sorry I don't see where, can you point me to it ?
And the ZBLANK column case is not tested (which is not surprising, astropy cannot write such a file).

Reply:
It is this one
https://github.com/astropy/astropy/blob/5c794f66638443263bcc4a04a87b7c438327a8c7/astropy/io/fits/hdu/compressed/compressed.py#L372-L377
and I think the case you mean is `cfile3` in the added test, manually constructed from the columns – one could argue though about testing the case _with_ a (custom) `BLANK` in the header as well.

Reply:
Thanks @dhomeier -- you are correct.  I included the `cfile3` test to get full coverage of the new code.  I suppose we could do additional tests if you think they would be useful.  They aren't needed for the code coverage though.

If I were to change this, I would probably choose a different value of `BLANK` for the tests that specify it (currently I'm using -32768, which is the same as the default that is supplied by the above code).

Reply:
> If I were to change this, I would probably choose a different value of `BLANK` for the tests that specify it (currently I'm using -32768, which is the same as the default that is supplied by the above code).

Yes, that would be exactly how I had it in mind, to confirm it is honouring the header value, and not just to keep codecov happy.

Reply:
Thanks! It had occurred to me that we should also test if the actual `data` for the invalid pixels are set to the `BLANK` value before conversion to float, but the compressed version does not have an `ignore_blank` option, and changing this is beyond the scope of this PR. Should keep that in mind for #15474 though (which may already bring it in automatically from `image.py`).
@pllim does the changelog check mean the changelog file has to be renamed or moved somewhere else, or do we simply ignore or `skip changelog-checks` for backports? I have unchecked the corresponding box in the list, but its meaning is not completely clear to me.

Reply:
Oh missed that somehow, sounds good then, thanks!